### DISCUSSION TITLE
Add RBAC Feature

### DISCUSSION THREAD

**Discussion by kanrio** [medusajs/medusa#4815] [Feature Requests]
I'm reaching out to propose a valuable feature addition that could significantly enhance MedusaJS's user management capabilities.
Tailoring user access to specific modules/APIs offers greater customization

Also what members and developer has access to ?

**Comment by alightinastorm** [comment]
RBAC can be highly domain specific, so it's probably much better to implement it on your side instead of Medusa?
I can't imagine the contributors having to maintain tons of rbac checks for every usecase

**Reply by kanrio** [reply]
They don't need to implement every usecase, just a place where you create a role and you check the possible modules/apis that this role can access like Strapi

**Reply by SalahAdDin** [reply]
yet there is no any plugin yet, I don't understand why.

**Reply by sbpro86** [reply]
@kanrio 
You are absolutely right. Did you figured out any solution to this ?

**Comment by radoslaw-sz** [comment]
hi @kanrio  - my company will release soon the RBAC functionality for v2. 

https://github.com/user-attachments/assets/704ac4a2-67ec-44da-8418-e40b80a07197

Unfortunately, we do not have a plan to make it open source (it will be behind the license). Anyway I am letting you know that you can reach me if interested.

**Reply by moaaz-bhnas** [reply]
Hi @radoslaw-sz, can you please share with us guidelines / insights of how you implemented this?

**Reply by SalahAdDin** [reply]
> Hi @radoslaw-sz, can you please share with us guidelines / insights of how you implemented this?

I don't think so, It will be a private plugin.
